scripts/jenkins/qa_openstack.sh

```diff
@@ -168,8 +168,10 @@ $zypper --gpg-auto-import-keys -n ref
 # deinstall some leftover crap from the cleanvm
 $zypper -n rm --force 'python-cheetah < 2.4'
 
-# deinstall cloud-init and dependencies
+# deinstall cloud-init and dependencies (but keep cloud-final systemd unit around)
```

Comment:
There's no bug report, no. I should probably submit one.

The situation in a nutshell: The `cloud-final` systemd unit specifies timeout=0, meaning no timeout for the cloud-init process. Once the unit file is removed by cloud-init deinstallation, systemd reverts to its default of setting a timeout (10 minutes if I remember correctly) and killing the process after that timeout has passed.

As for running `qa_openstack.sh` without heat: this doesn't matter much in that case since the cloud-final process will be long finished by that time. What I could do:

1) Make retention of the cloud-final service file dependent on an environment variable set by the user data script.
2) Remove that service file after at the end of qa_openstack.sh to make sure it doesn't remain as a potential future problem.

Would that be ok for you?